### PR TITLE
feat(getIsVehicleInitiallyLocked): integrate with lockNPCCarsChance

### PR DESCRIPTION
**Pull request by ArturMichalak**
## Description

Chance to lock a vehicle parked and driven by an npc.

## Checklist

- [X] I have personally loaded this code into an updated Qbox project and checked all of its functionality.
- [X] My pull request fits the contribution guidelines & code conventions.

### CODE REVIEW

File: config/shared.lua [hunk 1]
```diff
@@ -10,7 +10,8 @@ return {
         ---@type VehicleConfig
         default = {
             noLock = false,
-            spawnLocked = 1.0,
+            spawnLocked = .75,
```
Comment (Manason) [inline config/shared.lua:13]: Maybe rename to spawnLockedIfParked?

File: shared/functions.lua [hunk 2]
```diff
@@ -35,13 +35,18 @@ end
...
+    local vehicleConfig = public.getVehicleConfig(vehicle)
+    local vehicleLockedChance = isDriven
+        and vehicleConfig.drivenSpawnLocked
+        or vehicleConfig.spawnLocked
+
+    if type(vehicleLockedChance) == 'number' then
+        return math.random() < vehicleLockedChance
     else
-        return isVehicleSpawnLocked ~= nil
+        return vehicleLockedChance ~= nil
```
Comment (Manason) [inline shared/functions.lua:49]:
```suggestion
        return vehicleLockedChance == true
```
Needs to be an explicit true check as false is a valid input

**Review by Manason — CHANGES_REQUESTED** [review]

File: server/main.lua [hunk 3]
```diff
@@ -64,19 +63,13 @@ AddEventHandler('entityCreated', function (entity)
...
 
     local type = GetEntityType(entity)
-
-    if type ~= EntityType.Ped and type ~= EntityType.Vehicle then
-        return
-    end
-
-    local vehicle = type == EntityType.Ped and GetVehiclePedIsIn(entity, false) or entity
+    local isPed = type == EntityType.Ped
+    if not (isPed or type == EntityType.Vehicle) then return end
```
Comment (Manason) [inline server/main.lua:67]: I think the old syntax here was more readable than pulling out the not.

**Review by Manason — APPROVED** [review]

File: server/main.lua (hunk 3 above)
Comment (ArturMichalak) [reply server/main.lua:67]: what do you think about
```lua
    local isPed = type == EntityType.Ped
    local isVehicle = type == EntityType.Vehicle
    if not (isPed or isVehicle) then return end
```

Comment (Manason) [reply server/main.lua]: I think it's less cognitive overhead to be written as `not isPed and not isVehicle`, but if you want to keep it this way it's not a big deal.

**Pull request merged**